corehq/motech/repeaters/models.py

```diff
@@ -188,6 +189,44 @@ def log_repeater_success_in_datadog(domain, status_code, repeater_type):
 )
 
 
+class RepeaterSuperProxy(models.Model):
+    # See https://stackoverflow.com/questions/241250/single-table-inheritance-in-django/60894618#60894618
+    PROXY_FIELD_NAME = "repeater_type"
+
+    repeater_type = models.CharField(max_length=64)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        # _repeater_type should be replaces when SQL is removed from model name
+        # After that self.model.__name__ would work fine
+        self.repeater_type = self._repeater_type
```

Comment:
What's the difference between `self.model.__name__` and `self.__class__.__name__`? 

(As an aside, I'm also curious about `self.__class__.__name__` (which I've seen many times) and `type(self).__name__` (which I noticed in the SO answer).)

It might be quite easy to avoid `self._repeater_type`:

Naive alternative:
```python
>>> def get_repeater_type(repeater):
...     name = repeater.__class__.__name__
...     return name.lstrip('SQL')
...
>>> get_repeater_type(SQLFooRepeater())
'FooRepeater'
>>> get_repeater_type(FooRepeater())
'FooRepeater'
>>> get_repeater_type(SQLQuxRepeater())
'uxRepeater'
>>> get_repeater_type(QuxRepeater())
'uxRepeater'
```

Better alternative:
```python
>>> def get_repeater_type(repeater):
...     name = repeater.__class__.__name__
...     return name[:3] if name.startswith('SQL') else name
...
>>> get_repeater_type(SQLQuxRepeater())
'QuxRepeater'
>>> get_repeater_type(QuxRepeater())
'QuxRepeater'
```

Reply:
Thanks a lot, @kaapstorm  for suggesting the approach for getting the repeater type, it surely is 10X cleaner than what I have. 

> What's the difference between self.model.__name__ and self.__class__.__name__?

Since we need to know the actual model class inside the manager so we are using `self.model.__name__`
As per Django [documentation](https://docs.djangoproject.com/en/3.2/topics/db/managers/#adding-extra-manager-methods) 
```
Another thing to note is that Manager methods can access self.model to get the model class to which they’re attached.
``` 

whereas `self.__class__` refers to the class to which `self` refers. 

> (As an aside, I'm also curious about self.__class__.__name__ (which I've seen many times) and type(self).__name__ (which I noticed in the SO answer).)

In my understanding, both should be the same. I looked it up and found [this](https://stackoverflow.com/questions/1060499/difference-between-typeobj-and-obj-class) answer on StackOverflow which basically says they are mostly the same in Python 3 unless you override `__class__` method.